base_edifact/models/edifact.py

```diff
@@ -199,7 +199,7 @@ def map2odoo_product(self, seg, pia=None):
             ['5', ['1276', 'SA', '', '9']]
         SA. Supplier's Article Number
         """
-        code = seg[2][0]
+        code = seg[2][0] if len(list(seg)) > 2 else False
```

Comment:
no need to convert to list
```suggestion
        code = seg[2][0] if len(seg) > 2 else False
```

Reply:
Sorry @simahawk , maybe we need to convert to `list` to compare the elements
```Traceback (most recent call last):
  File "/odoo/src/odoo/tools/misc.py", line 818, in deco
    return func(*args, **kwargs)
  File "/odoo/local-src/allo_edi_sale/tests/test_edifact_order_import.py", line 112, in test_edifact_order_import_incorrect_barcode
    wiz.import_order_button()
  File "/odoo/external-src/edi/sale_order_import/wizard/sale_order_import.py", line 366, in import_order_button
    parsed_order = self.parse_order(
  File "/odoo/external-src/edi/sale_order_import/wizard/sale_order_import.py", line 342, in parse_order
    parsed_order = self._parse_file(order_filename, order_file)
  File "/odoo/external-src/edi/sale_order_import/wizard/sale_order_import.py", line 105, in _parse_file
    return getattr(self, "parse_%s_order" % self.import_type)(
  File "/odoo/external-src/edi/sale_order_import_edifact/wizard/sale_order_import.py", line 120, in parse_edifact_order
    lines = self._prepare_edifact_lines(interchange)
  File "/odoo/external-src/edi/sale_order_import_edifact/wizard/sale_order_import.py", line 262, in _prepare_edifact_lines
    "product": edifact_model.map2odoo_product(linseg, piaseg),
  File "/odoo/external-src/edi/base_edifact/models/edifact.py", line 202, in map2odoo_product
    code = seg[2][0] if len(seg) > 2 else False
TypeError: object of type 'Segment' has no len()
```

Reply:
ah sorry... `Segment` implements slicing but not `__len__`... cool :rofl: